proj/src/main/webapp/chat.js

```diff
@@ -283,8 +282,8 @@ async function setUserTags(tagList){
 async function addUserTags(tagList){
     if(firebase.auth().currentUser){
 
-        const abridgedTagsRef = "/users/" + firebaseUser.uid + "/allTags";
-        const abridgedTagRemovalRef = "/users/" + firebaseUser.uid + "/tagRemovalDict";
+        const abridgedTagsRef = "/users/" + currentUID + "/allTags";
+        const abridgedTagRemovalRef = "/users/" + currentUID + "/tagRemovalDict";
```

Comment:
I'm not sure where `currentUID` is defined. Also having it defined globally, isn't a great idea. Can we replace it wth `firebase.auth().currentUser.uid` everywhere it is used so that functions are not dependent on the global variable wherever it is set.

Reply:
fixed